Unable to successfully execute su root

**Describe the bug:**
I tried running `su root` in `murex`, but after entering my password, it freezes completely with no response. However, `Ctrl+C` can still terminate it.

Additionally, I have taken the following steps:
- Deleted all my personal murex profiles
- Downloaded the latest version of murex

Neither worked.

Below is the output of running `murex -debug`:
```
Loading profile `.murex_preload`
2026/04/07 11:53:29 Loading profile `builtin/profile_preload`
Error in `f` ((builtin) 1,44):
      Command: f +x
      Error: no data returned
Error in `f` ((builtin) 1,250):
      Command: f +x
      Error: no data returned
2026/04/07 11:53:29 Loading profile `builtin/profile_any`
2026/04/07 11:53:29 Loading profile `builtin/profile_linux`
Error in `expr` ((builtin) 1,988):
      Command: $WSLENV
      Error: variable 'WSLENV' does not exist
Error in `expr` ((builtin) 1,988):
      Command: $WSL_DISTRO_NAME
      Error: variable 'WSL_DISTRO_NAME' does not exist
2026/04/07 11:53:29 Loading profile `builtin/profile_posix`
2026/04/07 11:53:29 Loading profile `builtin/profile`
2026/04/07 11:53:29 Loading profile `builtin/integrations_su_linux`
2026/04/07 11:53:29 Loading profile `builtin/integrations_systemctl_linux`
2026/04/07 11:53:29 Loading profile `builtin/integrations_yay_linux`
2026/04/07 11:53:29 Loading profile `builtin/integrations_zzz_example_linux`
2026/04/07 11:53:29 Loading profile `builtin/integrations_bash_posix`
2026/04/07 11:53:29 Loading profile `builtin/integrations_coreutils_posix`
panic: More closed dependents than open

goroutine 673 [running]:
github.com/lmorg/murex/builtins/pipes/streams.init.func1(0x1e?)
	/home/runner/work/murex/murex/builtins/pipes/streams/neg_panic.go:7 +0x34
github.com/lmorg/murex/builtins/pipes/streams.(*Stdin).Close(0xc0001f6d90)
	/home/runner/work/murex/murex/builtins/pipes/streams/define.go:75 +0x4f
github.com/lmorg/murex/lang.UnmarshalDataBuffered(0x40ba08?, {0xc00061aea0, 0xe, 0x10}, {0xc38dd1, 0x4})
	/home/runner/work/murex/murex/lang/define_unmarshal.go:60 +0x1e7
github.com/lmorg/murex/lang.(*Variables).getEnvValueValue(0xc0001d07c0, {0xc0000280f0, 0x4}, {0xc0000280f5, 0xe})
	/home/runner/work/murex/murex/lang/variables.go:277 +0x191
github.com/lmorg/murex/lang.getEnvVarValue(0xc0001d07c0)
	/home/runner/work/murex/murex/lang/variables_reserved.go:71 +0xcc
github.com/lmorg/murex/lang.(*Variables).getValue(0xc0001d07c0, {0xc00061ae90?, 0xe52de8?})
	/home/runner/work/murex/murex/lang/variables.go:161 +0x8d
github.com/lmorg/murex/lang.(*Variables).Set(0xc0001d07c0, 0xc0000cfc60, {0xc00061ae90, 0xc}, {0xb7eba0, 0xe716c8}, {0xc38d3d, 0x4})
	/home/runner/work/murex/murex/lang/variables.go:597 +0xde
github.com/lmorg/murex/lang/expressions.(*ParserT).setVar(...)
	/home/runner/work/murex/murex/lang/expressions/variables.go:108
github.com/lmorg/murex/lang/expressions.expAssign(0xc000126a50, 0x1)
	/home/runner/work/murex/murex/lang/expressions/exp14.go:145 +0x566
github.com/lmorg/murex/lang/expressions.executeExpression(0xc000126a50, 0x15)
	/home/runner/work/murex/murex/lang/expressions/expression.go:93 +0x7f
github.com/lmorg/murex/lang/expressions.(*ParserT).executeExpr(0xc000126a50)
	/home/runner/work/murex/murex/lang/expressions/expression.go:18 +0x85
github.com/lmorg/murex/lang/expressions.ExecuteExpr(0xc0000cfc60, {0xc0001661e0, 0x15, 0x18})
	/home/runner/work/murex/murex/lang/expressions/execute.go:16 +0xae
github.com/lmorg/murex/builtins/core/expressions.cmdExpressions(0xc0000cfc60)
	/home/runner/work/murex/murex/builtins/core/expressions/expressions.go:16 +0x45
github.com/lmorg/murex/lang.executeProcess(0xc0000cfc60)
	/home/runner/work/murex/murex/lang/process.go:332 +0x129f
created by github.com/lmorg/murex/lang.runModeNormal in goroutine 1
	/home/runner/work/murex/murex/lang/interpreter_pc.go:47 +0x3f
```

**Platform (please complete the following information):**
 - Linux 6.12.74+deb13+1-amd64 Debian 6.12.74-2 (2026-03-08) x86_64 GNU/Linux
 - Terminal: Rio
 - murex v7.2.1001 (unknown)

--- a/lang/define_unmarshal.go
+++ b/lang/define_unmarshal.go
@@ -47,8 +47,10 @@ func UnmarshalDataBuffered(parent *Process, b []byte, dataType string) (any, err
 	fork := parent.Fork(F_BACKGROUND | F_CREATE_STDIN | F_NO_STDOUT | F_NO_STDERR)
 	defer fork.Kill()
 
-	_, err := fork.Stdin.Write(b)
+	fork.Stdin.Open()
 	defer fork.Stdin.Close()
+
+	_, err := fork.Stdin.Write(b)
 	if err != nil {
 		return nil, fmt.Errorf("cannot write value to unmarshaller's buffer: %s", err.Error())
 	}

--- a/lang/define_unmarshal.go
+++ b/lang/define_unmarshal.go
@@ -48,12 +48,12 @@ func UnmarshalDataBuffered(parent *Process, b []byte, dataType string) (any, err
 	defer fork.Kill()
 
 	fork.Stdin.Open()
-	defer fork.Stdin.Close()
-
 	_, err := fork.Stdin.Write(b)
+	fork.Stdin.Close()
 	if err != nil {
 		return nil, fmt.Errorf("cannot write value to unmarshaller's buffer: %s", err.Error())
 	}
+
 	v, err := UnmarshalData(fork.Process, dataType)
 	if err != nil {
 		return nil, fmt.Errorf("cannot unmarshal buffer: %s", err.Error())

--- a/lang/define_unmarshal_test.go
+++ b/lang/define_unmarshal_test.go
@@ -5,10 +5,32 @@ import (
 	"testing"
 
 	_ "github.com/lmorg/murex/builtins"
+	"github.com/lmorg/murex/debug"
 	"github.com/lmorg/murex/lang"
 	"github.com/lmorg/murex/lang/types"
 	"github.com/lmorg/murex/test/count"
 )
+
+func TestUnmarshalDataBufferedDebug(t *testing.T) {
+	count.Tests(t, 1)
+
+	input := `["a","b","c"]`
+
+	lang.InitEnv()
+
+	debug.Enabled = true
+	defer func() { debug.Enabled = false }()
+
+	v, err := lang.UnmarshalDataBuffered(lang.ShellProcess, []byte(input), types.Json)
+	if err != nil {
+		t.Error(err)
+		return
+	}
+
+	if fmt.Sprintf("%v", v) != "[a b c]" {
+		t.Errorf("unexpected output: %v", v)
+	}
+}
 
 func TestUnmarshalArrayJsonString(t *testing.T) {
 	count.Tests(t, 1)

